remove change-id assertion in bootstrap test

onur-ozkan opened the pull request:
In the bootstrap test, the assertion of the change-id fails whenever we update the change-id next to a breaking change in build configurations. This commit removes the assertion, as it's not critical or useful to have.


ref https://github.com/rust-lang/rust/pull/115898#issuecomment-1775909050

onur-ozkan commented:
r? albertlarsan68 (Mark has too many PRs assigned already)

bors commented:
:umbrella: The latest upstream changes (presumably #116998) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

albertlarsan68 reviewed (approved):
r=me when rebased, although I would prefer if there was a way to check that the value is what is expected.

onur-ozkan committed:
remove change-id assertion in bootstrap test

In the bootstrap test, the assertion of the change-id
fails whenever we update the change-id next to a breaking change
in build configurations. This commit removes the assertion,
as it's not critical or useful to have.

Signed-off-by: onur-ozkan <work@onurozkan.dev>

onur-ozkan commented:
> although I would prefer if there was a way to check that the value is what is expected.

Testing `change-id` is unnecessary since we already have sanity checks for it. Additionally, testing a dynamic value with a hard-coded one is not a good practice.

@bors r=albertlarsan68 rollup

bors commented:
:pushpin: Commit b99b93586fe949f87f20cc65b9f0f1ac738c529d has been approved by `albertlarsan68`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=albertlarsan68 b99b93586fe949f87f20cc65b9f0f1ac738c529d -->
<!-- homu: {"type":"Approved","sha":"b99b93586fe949f87f20cc65b9f0f1ac738c529d","approver":"albertlarsan68","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
